### ISSUE TITLE
Asciidoctor v2.x.x

### ISSUE THREAD

**Issue by ghost** [owenh000/asciidoctor-multipage#2]
Hey there, 

are there any plans to get this working with newer Asciidoctor versions? 
I have tried to do it myself but I am not familiar with the Asciidoctor code base and so far I've come up empty.

Thanks.

**Comment by owenh000** [comment]
@ldz-w, it would be great to get this extension working with current Asciidoctor versions. However I am not able to work on this more unless someone is interested in sponsoring.

If you are looking at the code, please note the Ruby API documentation at <https://www.rubydoc.info/gems/asciidoctor> is very helpful, but see <https://github.com/asciidoctor/asciidoctor/issues/3554>.

**Comment by woj-tek** [comment]
>  it would be great to get this extension working with current Asciidoctor versions.

What are the main roadblocks here (just hints / general pointers)?

**Comment by owenh000** [comment]
Hi @woj-tek, thank you for your interest in asciidoctor-multipage. A start would be to run the sample file as shown in the README but with a newer version of Asciidoctor. The following example adds `--trace` to show a full backtrace for in case of an error:

```
$ gem install --user-install asciidoctor
$ gem list asciidoctor$
asciidoctor (2.0.10, 1.5.7.1)
$ asciidoctor --trace -r ./asciidoctor-multipage.rb -b multipage_html5 \
      test/sample.adoc
Traceback (most recent call last):
	16: from /home/user/.gem/ruby/2.5.0/bin/asciidoctor:23:in `<main>'
	15: from /home/user/.gem/ruby/2.5.0/bin/asciidoctor:23:in `load'
	14: from /home/user/.gem/ruby/2.5.0/gems/asciidoctor-2.0.10/bin/asciidoctor:15:in `<top (required)>'
	13: from /home/user/.gem/ruby/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/cli/invoker.rb:111:in `invoke!'
	12: from /home/user/.gem/ruby/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/cli/invoker.rb:111:in `each'
	11: from /home/user/.gem/ruby/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/cli/invoker.rb:128:in `block in invoke!'
	10: from /home/user/.gem/ruby/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/convert.rb:183:in `convert_file'
	 9: from /home/user/.gem/ruby/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/convert.rb:183:in `open'
	 8: from /home/user/.gem/ruby/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/convert.rb:183:in `block in convert_file'
	 7: from /home/user/.gem/ruby/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/convert.rb:118:in `convert'
	 6: from /home/user/.gem/ruby/2.5.0/gems/asciidoctor-2.0.10/lib/asciidoctor/document.rb:951:in `convert'
	 5: from /home/user/asciidoctor-multipage.git/asciidoctor-multipage.rb:120:in `convert'
	 4: from /home/user/asciidoctor-multipage.git/asciidoctor-multipage.rb:170:in `document'
	 3: from /home/user/asciidoctor-multipage.git/asciidoctor-multipage.rb:354:in `new_outline_doc'
	 2: from /home/user/asciidoctor-multipage.git/asciidoctor-multipage.rb:354:in `each'
	 1: from /home/user/asciidoctor-multipage.git/asciidoctor-multipage.rb:355:in `block in new_outline_doc'
/home/user/asciidoctor-multipage.git/asciidoctor-multipage.rb:370:in `new_outline_doc': undefined method `number=' for #<Asciidoctor::Section:0x000055e05ac91090> (NoMethodError)
Did you mean?  number
               numbered=
               numbered
```

So there is an `Asciidoctor::Section` method missing. We need to fix that by using another method or otherwise adapting to the change in Asciidoctor. Then repeat with the next error.

Note, ideally we would have a test suite that checks compatibility with each supported version of Asciidoctor; we have issue #4 for this.

I hope this helps. Thank you!

**Comment by owenh000** [comment]
Asciidoctor v2.0.12 is supported as of commit 07a4da532f1f942929b572ba29a21a04f77e7cd3. The changes are included in the v0.0.2 release.

Closing this issue.
